Fixes QRCode style

ArthurHoaro opened the pull request:
* fixes a regression misplacing QRCode popup introduced in 6a6f6c32e5876a347a806cc43d25596b9257765b (`position: relative` was missing).
 * adds a fancy transition on popup openning from @alexisju [proposition](https://github.com/shaarli/Shaarli/commit/6a6f6c32e5876a347a806cc43d25596b9257765b#commitcomment-14943549).

alexisju commented:
It work well. 

But still another css issue? In my test, i go different box size : qr is not always centered in the box 
http://effingo.be/test/shaarli

Btw my suggestion was not about the "fancy" effect, but about the centred qrcode with shadowed backgroud.

What i dont like in the current way (with ou without fancy effect) is that qr popups are displayed at relative positions ; sometime it look weird or not displayed totaly (if page is scrolled down too much) ; relative positions are unnecessary and not the much convenient to catch with a smartphone (just centered is perfect, and graphically nicer).. But it is very personal appreciation... Don't worry about that ;)

ArthurHoaro commented:
I'll probably remove the effect. After sleeping on it, I don't think we should add effects that adds loading time. But it was fun. :) 

I like the way it pops where you click. It's very subjective for sure. But you can override it the way you want in your theme. 

About the centered issue, I experienced it, but it's not a bug. It's just white blocks in the image. I'll see what I can do. 

I didn't test it on mobile, my bad.

ArthurHoaro commented:
As I said:

  * removed CSS transition effect.
  * added a border to the image for blank line and center 'issue'.
  * for mobile:
     * it's not very pretty but it works.
     * QRCode are usually used for PC to mobile.

alexisju commented:
The box-shadow around the qr image looks weirds...

alexisju commented:
Just for information, i pushed a new plugin `qrcode-albinomouse` here https://github.com/alexisju/qrcode_albinomouse/  demo : https://liens.effingo.be/

ArthurHoaro commented:
It looks great, but it looks like you only changed the CSS. Why didn't you just changed your theme CSS instead of creating a plugin?

alexisju commented:
To separate template and plugin management. You can only want to install one of both.

(btw they are also two minor changes in shaarli-qrcode.js : bigger shortcode img and commented message "click to close").

virtualtam commented on plugins/qrcode/shaarli-qrcode.js:
```diff
@@ -37,6 +37,12 @@ function showQrCode(caller,loading)
+        // Show the QRCode
+        qrcodeImage = document.getElementById('permalinkQrcode');
+        // Workaround to deal with newly created element lag for transition.
+        window.getComputedStyle(qrcodeImage).opacity;
```
seems like this value is not used

ArthurHoaro replied:
Look at the comment, it's a (hacky) way to actually execute the CSS transition with the show class, from [SO thread](http://stackoverflow.com/questions/12088819/css-transitions-on-new-elements). Also it seems way more reliable than `setTimout()`.

virtualtam replied:
Thanks for the link :)
